### ISSUE TITLE
Modifying shellcode 

### ISSUE THREAD

**Issue by poopaapoopaa** [EmpireProject/Empire#1200]
Probably a stupid question, but where is the source of shellcodes? I want to try to modify the shells (for example of launcher_bat or launcher_vbs) in their pre-encoded form, in an attempt to evade Defender . 

I went through the directories and couldn't figure out where they are stored.

**Comment by xorrior** [comment]
@poopaapoopaa are you referring to the actual shellcode stagers or to the bat and vbs launchers?

**Comment by poopaapoopaa** [comment]
I'm referring to the encoded parts in the bat / vbs launchers. After more searching I found where it gets built. It's in the listeners code, for example in:
/Empire/lib/listeners/http.py - generate_launcher()

Motivation: I added -NoExit the launcher.bat script to try to identify how defender detects it.
I got the error: 

```
At line:1 char:1
+ IF($PSVeRsIOnTabLE.PSVERSIoN.MaJOR -gE 3){$GPF=[reF].ASsemBly.GeTTypE ..
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
This script contains malicious content and has been blocked by your antivirus software.
   + CategoryInfo              : ParserError: (:) [], ParentContainsErrorRecordException
   + FullyQualifiesErrorId   : ScriptContainedMaliciousContent
```
This indicates defender checks the decoded version. I wonder if trying to add code at the beginning of the shellcode will help here. Perhaps dump the stager string (pre-encoded) to a file let defender examine chunks of it to find the problem. 

I realize AV detection isn't a direct issue with Empire but will still appreciate any ideas here.
